src/backend/drakula/database.py

```
@@ -1,74 +1,96 @@
+import uuid
+from dotenv import load_dotenv
 from os import getenv
+import os
 from typing import Optional
 from hashlib import md5
-
-from mysql.connector import connect
-
+from mysql.connector import connect, Error
 from .models import Airport
 
 DEFAULT_AIRPORT_AMOUT = 15
 
+load_dotenv()
+
 
 class Database:
     def __init__(self, user, password, host, port):
-        self.connection = connect(
-            host=host,
-            port=port,
-            user=user,
-            password=password,
-            database="flight_game",
-            autocommit=True,
-        )
+        try:
+            self.connection = connect(
+                host=host,
+                port=port,
+                user=user,
+                password=password,
+                database="flight_game",
+                autocommit=True,
+                auth_plugin="mysql_native_password",
+            )
+        except Error as e:
```

I would encourage not handling this error and letting the program crash. This functionally silences the exception and break the invariant of the Database class. Now, instead of having the program just die if the database doesn't work, we introduce a state where the Database class exists, but does not have a connection, so we are forced to check nullability at call site everywhere.

---

ok

---

done